### ISSUE TITLE
DataStore - Failed to publish a local change to DynamoDB.

### ISSUE THREAD

**Issue by upachecog** [aws-amplify/amplify-android#1244]
**Description:** 

My local changes are not being synchronized to DynamoDB. 

**Amplify Version**

- I was able to trace the bug back to version 1.6.11.
- DataStore version 1.6.9 is working fine with the same models. 👍
- I am currently using DataStore version 1.17.1 and the error is still present.

**Code:** 

With the following code, I am trying to update the state of my model called Actividad (Activity).

```
  override fun updateActivityState(
      activityState: ActivityState,
      activityId: String,
      image: String?,
      onResponse: (value: Activity) -> Unit
  ) {
      activityDataStore.updateActivity(
          transformActivityStateToEstatusActividad(activityState),
          activityId,
          image,
          { actividad ->
              Log.v(TAG, "Updated Activity $activityId state to $activityState")
              val updatedActivity = activityAdapter.transformActividadToActivity(actividad)
              onResponse(updatedActivity)
          },
          {
              Log.v(TAG, "Failed to update activity state")
          })
  }
```

4. Even though my local DB is being updated, I cannot see those changes in Dynamo and the log shows the following error:

```
V/ActivityUpdater: Updated Activity 8265aeb7-0e2d-44b0-a999-7ea27d135625 state to COMPLETED
I/amplify:aws-datastore: Successfully enqueued PendingMutation{mutatedItem=SerializedModel{id='8265aeb7-0e2d-44b0-a999-7ea27d135625', serializedData={imagen=, id=8265aeb7-0e2d-44b0-a999-7ea27d135625, estatus=TERMINADO, updatedAt=2021-03-22T07:26:31.043Z}, modelSchema=ModelSchema{name='Actividad', pluralName='Actividads', authRules=[AuthRule{authStrategy=OWNER, ownerField='asignado', identityClaim='cognito:username', groupsField='', groupClaim='', groups=[]', operations=[READ, UPDATE]'}, AuthRule{authStrategy=GROUPS, ownerField='', identityClaim='', groupsField='', groupClaim='cognito:groups', groups=[admin]', operations=[READ, CREATE, UPDATE, DELETE]'}], fields={agrupacion=ModelField{name='agrupacion', javaClassForValue='class java.lang.Integer', targetType='Int', isRequired=false, isArray=false, isEnum=false, isModel=false}, asignado=ModelField{name='asignado', javaClassForValue='class java.lang.String', targetType='String', isRequired=true, isArray=false, isEnum=false, isModel=false}, coordenada=ModelField{name='coordenada', javaClassForValue='interface java.util.List', targetType='String', isRequired=true, isArray=true, isEnum=false, isModel=false}, data=ModelField{name='data', javaClassForValue='class java.lang.String', targetType='String', isRequired=false, isArray=false, isEnum=false, isModel=false}, descripcion=ModelField{name='descripcion', javaClassForValue='class java.lang.String', targetType='String', isRequired=true, isArray=false, isEnum=false, isModel=false}, estatus=ModelField{name='estatus', javaClassForValue='class com.amplifyframework.datastore.generated.model.EstatusActividad', targetType='EstatusActividad', isRequired=true, isArray=false, isEnum=true, isModel=false}, fecha=ModelField{name='fecha', javaClassForValue='class com.amplifyframework.core.model.temporal.Temporal$Date', targetType='AWSDate', isRequired=true, isArray=false, isEnum=false, isModel=false}, id=ModelField{name='id', javaClassForValue='class java.lang.String', targetType='ID', isRequired=true, isArray=false, isEnum=false, isModel=false}, imagen=ModelField{name='imagen', javaClassForValue='class java.lang.String', targetType='String', isRequired=false, isArray=false, isEnum=false, isModel=false}, orden=ModelField{name='orden', javaClassForValue='class java.lang.Integer', targetType='Int', isRequired=true, isArray=false, isEnum=false, isModel=false}, tipo=ModelField{name='tipo', javaClassForValue='class com.amplifyframework.datastore.generated.model.Tipo', targetType='Tipo', isRequired=true, isArray=false, isEnum=true, isModel=false}, ubicacion=ModelField{name='ubicacion', javaClassForValue='class com.amplifyframework.datastore.generated.model.Ubicacion', targetType='Ubicacion', isRequired=false, isArray=false, isEnum=false, isModel=true}, updatedAt=ModelField{name='updatedAt', javaClassForValue='class com.amplifyframework.core.model.temporal.Temporal$DateTime', targetType='AWSDateTime', isRequired=true, isArray=false, isEnum=false, isModel=false}}, associations={ubicacion=ModelAssociation{name='BelongsTo', targetName='ubicacion_id', associatedName='null', associatedType='Ubicacion'}}, indexes={}, modelClass=class com.amplifyframework.datastore.generated.model.Actividad}}, modelSchema=ModelSchema{name='Actividad', pluralName='Actividads', authRules=[AuthRule{authStrategy=OWNER, ownerField='asignado', identityClaim='cognito:username', groupsField='', groupClaim='', groups=[]', operations=[READ, UPDATE]'}, AuthRule{authStrategy=GROUPS, ownerField='', identityClaim='', groupsField='', groupClaim='cognito:groups', groups=[admin]', operations=[READ, CREATE, UPDATE, DELETE]'}], fields={agrupacion=ModelField{name='agrupacion', javaClassForValue='class java.lang.Integer', targetType='Int', isRequired=false, isArray=false, isEnum=false, isModel=false}, asignado=ModelField{name='asignado', javaClassForValue='class java.lang.String', targetType='String', isRequired=true, isArray=false, isEnum=false, isModel=false}, coordenada=ModelField{name='coordenada', javaClassForValue='interface java.util.List', targetType='String'
W/amplify:aws-datastore: Failed to publish a local change = PendingMutation{mutatedItem=SerializedModel{id='8265aeb7-0e2d-44b0-a999-7ea27d135625', serializedData={imagen=, id=8265aeb7-0e2d-44b0-a999-7ea27d135625, estatus=TERMINADO, updatedAt=2021-03-22T07:26:31.043Z}, modelSchema=ModelSchema{name='Actividad', pluralName='Actividads', authRules=[AuthRule{authStrategy=OWNER, ownerField='asignado', identityClaim='cognito:username', groupsField='', groupClaim='', groups=[]', operations=[READ, UPDATE]'}, AuthRule{authStrategy=GROUPS, ownerField='', identityClaim='', groupsField='', groupClaim='cognito:groups', groups=[admin]', operations=[READ, CREATE, UPDATE, DELETE]'}], fields={agrupacion=ModelField{name='agrupacion', javaClassForValue='class java.lang.Integer', targetType='Int', isRequired=false, isArray=false, isEnum=false, isModel=false}, asignado=ModelField{name='asignado', javaClassForValue='class java.lang.String', targetType='String', isRequired=true, isArray=false, isEnum=false, isModel=false}, coordenada=ModelField{name='coordenada', javaClassForValue='interface java.util.List', targetType='String', isRequired=true, isArray=true, isEnum=false, isModel=false}, data=ModelField{name='data', javaClassForValue='class java.lang.String', targetType='String', isRequired=false, isArray=false, isEnum=false, isModel=false}, descripcion=ModelField{name='descripcion', javaClassForValue='class java.lang.String', targetType='String', isRequired=true, isArray=false, isEnum=false, isModel=false}, estatus=ModelField{name='estatus', javaClassForValue='class com.amplifyframework.datastore.generated.model.EstatusActividad', targetType='EstatusActividad', isRequired=true, isArray=false, isEnum=true, isModel=false}, fecha=ModelField{name='fecha', javaClassForValue='class com.amplifyframework.core.model.temporal.Temporal$Date', targetType='AWSDate', isRequired=true, isArray=false, isEnum=false, isModel=false}, id=ModelField{name='id', javaClassForValue='class java.lang.String', targetType='ID', isRequired=true, isArray=false, isEnum=false, isModel=false}, imagen=ModelField{name='imagen', javaClassForValue='class java.lang.String', targetType='String', isRequired=false, isArray=false, isEnum=false, isModel=false}, orden=ModelField{name='orden', javaClassForValue='class java.lang.Integer', targetType='Int', isRequired=true, isArray=false, isEnum=false, isModel=false}, tipo=ModelField{name='tipo', javaClassForValue='class com.amplifyframework.datastore.generated.model.Tipo', targetType='Tipo', isRequired=true, isArray=false, isEnum=true, isModel=false}, ubicacion=ModelField{name='ubicacion', javaClassForValue='class com.amplifyframework.datastore.generated.model.Ubicacion', targetType='Ubicacion', isRequired=false, isArray=false, isEnum=false, isModel=true}, updatedAt=ModelField{name='updatedAt', javaClassForValue='class com.amplifyframework.core.model.temporal.Temporal$DateTime', targetType='AWSDateTime', isRequired=true, isArray=false, isEnum=false, isModel=false}}, associations={ubicacion=ModelAssociation{name='BelongsTo', targetName='ubicacion_id', associatedName='null', associatedType='Ubicacion'}}, indexes={}, modelClass=class com.amplifyframework.datastore.generated.model.Actividad}}, modelSchema=ModelSchema{name='Actividad', pluralName='Actividads', authRules=[AuthRule{authStrategy=OWNER, ownerField='asignado', identityClaim='cognito:username', groupsField='', groupClaim='', groups=[]', operations=[READ, UPDATE]'}, AuthRule{authStrategy=GROUPS, ownerField='', identityClaim='', groupsField='', groupClaim='cognito:groups', groups=[admin]', operations=[READ, CREATE, UPDATE, DELETE]'}], fields={agrupacion=ModelField{name='agrupacion', javaClassForValue='class java.lang.Integer', targetType='Int', isRequired=false, isArray=false, isEnum=false, isModel=false}, asignado=ModelField{name='asignado', javaClassForValue='class java.lang.String', targetType='String', isRequired=true, isArray=false, isEnum=false, isModel=false}, coordenada=ModelField{name='coordenada', javaClassForValue='in
W/amplify:aws-datastore: terface java.util.List', targetType='String', isRequired=true, isArray=true, isEnum=false, isModel=false}, data=ModelField{name='data', javaClassForValue='class java.lang.String', targetType='String', isRequired=false, isArray=false, isEnum=false, isModel=false}, descripcion=ModelField{name='descripcion', javaClassForValue='class java.lang.String', targetType='String', isRequired=true, isArray=false, isEnum=false, isModel=false}, estatus=ModelField{name='estatus', javaClassForValue='class com.amplifyframework.datastore.generated.model.EstatusActividad', targetType='EstatusActividad', isRequired=true, isArray=false, isEnum=true, isModel=false}, fecha=ModelField{name='fecha', javaClassForValue='class com.amplifyframework.core.model.temporal.Temporal$Date', targetType='AWSDate', isRequired=true, isArray=false, isEnum=false, isModel=false}, id=ModelField{name='id', javaClassForValue='class java.lang.String', targetType='ID', isRequired=true, isArray=false, isEnum=false, isModel=false}, imagen=ModelField{name='imagen', javaClassForValue='class java.lang.String', targetType='String', isRequired=false, isArray=false, isEnum=false, isModel=false}, orden=ModelField{name='orden', javaClassForValue='class java.lang.Integer', targetType='Int', isRequired=true, isArray=false, isEnum=false, isModel=false}, tipo=ModelField{name='tipo', javaClassForValue='class com.amplifyframework.datastore.generated.model.Tipo', targetType='Tipo', isRequired=true, isArray=false, isEnum=true, isModel=false}, ubicacion=ModelField{name='ubicacion', javaClassForValue='class com.amplifyframework.datastore.generated.model.Ubicacion', targetType='Ubicacion', isRequired=false, isArray=false, isEnum=false, isModel=true}, updatedAt=ModelField{name='updatedAt', javaClassForValue='class com.amplifyframework.core.model.temporal.Temporal$DateTime', targetType='AWSDateTime', isRequired=true, isArray=false, isEnum=false, isModel=false}}, associations={ubicacion=ModelAssociation{name='BelongsTo', targetName='ubicacion_id', associatedName='null', associatedType='Ubicacion'}}, indexes={}, modelClass=class com.amplifyframework.datastore.generated.model.Actividad}, mutationType=UPDATE, mutationId=ec429ab7-8adf-11eb-b08f-c59a253397a4, predicate=MatchAllQueryPredicate}
    DataStoreException{message=Error encountered while creating model schema, cause=DataStoreException{message=Failed to get fields for model., cause=AmplifyException{message=An invalid field was provided. ubicacion is not present in Actividad, cause=java.lang.IllegalStateException: Associated data is not Model or Map., recoverySuggestion=Check if this model schema is a correct representation of the fields in the provided Object}, recoverySuggestion=Validate your model file.}, recoverySuggestion=See attached exception for more details}
        at com.amplifyframework.datastore.appsync.AppSyncClient.update(AppSyncClient.java:168)
        at com.amplifyframework.datastore.syncengine.MutationProcessor.lambda$null$11$MutationProcessor(MutationProcessor.java:273)
        at com.amplifyframework.datastore.syncengine.-$$Lambda$MutationProcessor$nNiLkImAuB-YHEmz4H9EBhUR0O0.publish(Unknown Source:11)
        at com.amplifyframework.datastore.syncengine.MutationProcessor.lambda$publishWithStrategy$16(MutationProcessor.java:315)
        at com.amplifyframework.datastore.syncengine.-$$Lambda$MutationProcessor$A3yLotY3cqHqJ-_1xZmS_awSsgU.subscribe(Unknown Source:4)
        at io.reactivex.rxjava3.internal.operators.single.SingleCreate.subscribeActual(SingleCreate.java:40)
        at io.reactivex.rxjava3.core.Single.subscribe(Single.java:4813)
        at io.reactivex.rxjava3.internal.operators.single.SingleFlatMap.subscribeActual(SingleFlatMap.java:37)
        at io.reactivex.rxjava3.core.Single.subscribe(Single.java:4813)
        at io.reactivex.rxjava3.internal.operators.single.SingleFlatMap$SingleFlatMapCallback.onSuccess(SingleFlatMap.java:85)
        at io.reactivex.rxjava3.internal.operators.single.SingleCreate$Emitter.onSuccess(SingleCreate.java:68)
W/amplify:aws-datastore:     at com.amplifyframework.datastore.syncengine.VersionRepository.lambda$null$0$VersionRepository(VersionRepository.java:60)
        at com.amplifyframework.datastore.syncengine.-$$Lambda$VersionRepository$Kmuu7iZPxyGP2VIH1c1HFJFJDKw.accept(Unknown Source:8)
        at com.amplifyframework.datastore.storage.sqlite.SQLiteStorageAdapter.lambda$query$4$SQLiteStorageAdapter(SQLiteStorageAdapter.java:376)
        at com.amplifyframework.datastore.storage.sqlite.-$$Lambda$SQLiteStorageAdapter$CxFeClFgMnCdkUir6dsRxPfXmiA.run(Unknown Source:10)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:459)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:764)
     Caused by: DataStoreException{message=Failed to get fields for model., cause=AmplifyException{message=An invalid field was provided. ubicacion is not present in Actividad, cause=java.lang.IllegalStateException: Associated data is not Model or Map., recoverySuggestion=Check if this model schema is a correct representation of the fields in the provided Object}, recoverySuggestion=Validate your model file.}
        at com.amplifyframework.datastore.appsync.AppSyncRequestFactory.buildUpdateRequest(AppSyncRequestFactory.java:153)
        at com.amplifyframework.datastore.appsync.AppSyncClient.update(AppSyncClient.java:165)
        	... 19 more
     Caused by: AmplifyException{message=An invalid field was provided. ubicacion is not present in Actividad, cause=java.lang.IllegalStateException: Associated data is not Model or Map., recoverySuggestion=Check if this model schema is a correct representation of the fields in the provided Object}
        at com.amplifyframework.datastore.appsync.AppSyncRequestFactory.extractFieldLevelData(AppSyncRequestFactory.java:361)
        at com.amplifyframework.datastore.appsync.AppSyncRequestFactory.getMapOfFieldNameAndValues(AppSyncRequestFactory.java:320)
        at com.amplifyframework.datastore.appsync.AppSyncRequestFactory.buildUpdateRequest(AppSyncRequestFactory.java:150)
        	... 20 more
     Caused by: java.lang.IllegalStateException: Associated data is not Model or Map.
        at com.amplifyframework.datastore.appsync.AppSyncRequestFactory.extractAssociateId(AppSyncRequestFactory.java:377)
        at com.amplifyframework.datastore.appsync.AppSyncRequestFactory.extractFieldLevelData(AppSyncRequestFactory.java:356)
        	... 22 more
W/amplify:aws-datastore: Error ended observation of mutation outbox: 
    java.lang.RuntimeException: DataStoreException{message=Error encountered while creating model schema, cause=DataStoreException{message=Failed to get fields for model., cause=AmplifyException{message=An invalid field was provided. ubicacion is not present in Actividad, cause=java.lang.IllegalStateException: Associated data is not Model or Map., recoverySuggestion=Check if this model schema is a correct representation of the fields in the provided Object}, recoverySuggestion=Validate your model file.}, recoverySuggestion=See attached exception for more details}
        at io.reactivex.rxjava3.internal.util.ExceptionHelper.wrapOrThrow(ExceptionHelper.java:46)
        at io.reactivex.rxjava3.internal.observers.BlockingMultiObserver.blockingAwait(BlockingMultiObserver.java:145)
        at io.reactivex.rxjava3.core.Completable.blockingAwait(Completable.java:1490)
        at com.amplifyframework.datastore.syncengine.MutationProcessor.drainMutationOutbox(MutationProcessor.java:117)
        at com.amplifyframework.datastore.syncengine.MutationProcessor.lambda$startDrainingMutationOutbox$1$MutationProcessor(MutationProcessor.java:101)
        at com.amplifyframework.datastore.syncengine.-$$Lambda$MutationProcessor$E9H2I0qolxIT6NJns5-WPOBrLCY.apply(Unknown Source:4)
        at io.reactivex.rxjava3.internal.operators.observable.ObservableFlatMapCompletableCompletable$FlatMapCompletableMainObserver.onNext(ObservableFlatMapCompletableCompletable.java:97)
        at io.reactivex.rxjava3.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.drainNormal(ObservableObserveOn.java:201)
        at io.reactivex.rxjava3.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.run(ObservableObserveOn.java:255)
        at io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:65)
        at io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:56)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:764)
     Caused by: DataStoreException{message=Error encountered while creating model schema, cause=DataStoreException{message=Failed to get fields for model., cause=AmplifyException{message=An invalid field was provided. ubicacion is not present in Actividad, cause=java.lang.IllegalStateException: Associated data is not Model or Map., recoverySuggestion=Check if this model schema is a correct representation of the fields in the provided Object}, recoverySuggestion=Validate your model file.}, recoverySuggestion=See attached exception for more details}
        at com.amplifyframework.datastore.appsync.AppSyncClient.update(AppSyncClient.java:168)
        at com.amplifyframework.datastore.syncengine.MutationProcessor.lambda$null$11$MutationProcessor(MutationProcessor.java:273)
        at com.amplifyframework.datastore.syncengine.-$$Lambda$MutationProcessor$nNiLkImAuB-YHEmz4H9EBhUR0O0.publish(Unknown Source:11)
        at com.amplifyframework.datastore.syncengine.MutationProcessor.lambda$publishWithStrategy$16(MutationProcessor.java:315)
        at com.amplifyframework.datastore.syncengine.-$$Lambda$MutationProcessor$A3yLotY3cqHqJ-_1xZmS_awSsgU.subscribe(Unknown Source:4)
        at io.reactivex.rxjava3.internal.operators.single.SingleCreate.subscribeActual(SingleCreate.java:40)
        at io.reactivex.rxjava3.core.Single.subscribe(Single.java:4813)
        at io.reactivex.rxjava3.internal.operators.single.SingleFlatMap.subscribeActual(SingleFlatMap.java:37)
        at io.reactivex.rxjava3.core.Single.subscribe(Single.java:4813)
        at io.reactivex.rxjava3.internal.operators.single.SingleFlatMap$SingleFlatMapCallback.onSuccess(SingleFlatMap.java:85)
        at io.reactivex.rxjava3.internal.operators.single.SingleCreate$Emitter.onSuccess(SingleCreate.java:68)
W/amplify:aws-datastore:     at com.amplifyframework.datastore.syncengine.VersionRepository.lambda$null$0$VersionRepository(VersionRepository.java:60)
        at com.amplifyframework.datastore.syncengine.-$$Lambda$VersionRepository$Kmuu7iZPxyGP2VIH1c1HFJFJDKw.accept(Unknown Source:8)
        at com.amplifyframework.datastore.storage.sqlite.SQLiteStorageAdapter.lambda$query$4$SQLiteStorageAdapter(SQLiteStorageAdapter.java:376)
        at com.amplifyframework.datastore.storage.sqlite.-$$Lambda$SQLiteStorageAdapter$CxFeClFgMnCdkUir6dsRxPfXmiA.run(Unknown Source:10)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:459)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        	... 3 more
     Caused by: DataStoreException{message=Failed to get fields for model., cause=AmplifyException{message=An invalid field was provided. ubicacion is not present in Actividad, cause=java.lang.IllegalStateException: Associated data is not Model or Map., recoverySuggestion=Check if this model schema is a correct representation of the fields in the provided Object}, recoverySuggestion=Validate your model file.}
        at com.amplifyframework.datastore.appsync.AppSyncRequestFactory.buildUpdateRequest(AppSyncRequestFactory.java:153)
        at com.amplifyframework.datastore.appsync.AppSyncClient.update(AppSyncClient.java:165)
        	... 19 more
     Caused by: AmplifyException{message=An invalid field was provided. ubicacion is not present in Actividad, cause=java.lang.IllegalStateException: Associated data is not Model or Map., recoverySuggestion=Check if this model schema is a correct representation of the fields in the provided Object}
        at com.amplifyframework.datastore.appsync.AppSyncRequestFactory.extractFieldLevelData(AppSyncRequestFactory.java:361)
        at com.amplifyframework.datastore.appsync.AppSyncRequestFactory.getMapOfFieldNameAndValues(AppSyncRequestFactory.java:320)
        at com.amplifyframework.datastore.appsync.AppSyncRequestFactory.buildUpdateRequest(AppSyncRequestFactory.java:150)
        	... 20 more
     Caused by: java. lang.IllegalStateException: Associated data is not Model or Map.
        at com.amplifyframework.datastore.appsync.AppSyncRequestFactory.extractAssociateId(AppSyncRequestFactory.java:377)
        at com.amplifyframework.datastore.appsync.AppSyncRequestFactory.extractFieldLevelData(AppSyncRequestFactory.java:356)
        	... 22 more

```

These are the relevant models of the schema I am using:

```
type Actividad
@model
@key(name: "porActividad", fields: ["ubicacion_id", "descripcion"])
@key(name: "actividadPorEstatus", fields: ["estatus", "updatedAt"], queryField: "actividadPorEstatus")
@auth(rules: [
    { allow: owner, ownerField: "asignado", operations: [read, update] }
    { allow: groups, groups: ["admin"], operations: [read, create, update, delete] }
]) {
    id: ID!
    ubicacion_id: ID!
    ubicacion: Ubicacion @connection(fields: ["ubicacion_id"])
    fecha: AWSDate!
    descripcion: String!
    tipo: Tipo!
    estatus: EstatusActividad!
    orden: Int!
    agrupacion: Int
    data: String
    imagen: String
    coordenada: [String!]!
    asignado: String!
    updatedAt: AWSDateTime!
}

type Ubicacion
@model
@key(name: "porNombre", fields: ["nombre", "createdAt"], queryField: "porNombre")
@auth(rules: [
    { allow: owner, ownerField: "asignado", operations: [read, update] }
    { allow: groups, groups: ["admin"], operations: [read, create, update, delete] }
]){
    id: ID!
    fecha: AWSDate!
    fechaVisita: AWSDateTime
    sucursalId: String!
    ruta: String!
    nombre: String!
    codigoPostal: String
    ciudad: String
    entidadFederativa: String
    plaza: String
    proyecto: String
    cadena: String
    tipoSucursal: String
    perfil: String
    usuario: String
    municipio: String
    colonia: String
    calle: String
    numeroExterior: String
    numeroInterior: String
    entreCalles: String
    estatus: EstatusUbicacion!
    pais: String
    clasificacion: [String]
    coordenada: [String!]!
    imagen: [String]
    telefono: [AWSPhone]
    etiqueta: [String]
    distanciaMinima: Int
    asignado: String!
    createdAt: AWSDateTime!
    actividades: [Actividad] @connection(keyName: "porActividad", fields: ["id"])
}

enum EstatusUbicacion { }

enum EstatusActividad { }

enum Tipo { }

enum Evento { }

```

**Comment by raphkim** [comment]
Hi @upachecog , Thank you for the detailed report! I will take a look.

**Comment by richardmcclellan** [comment]
Potentially similar issue: https://github.com/aws-amplify/amplify-android/issues/1199

**Comment by haverchuck** [comment]
Potentiall [related issue ](https://github.com/aws-amplify/amplify-flutter/issues/463)in amplify flutter repo.

**Comment by raphkim** [comment]
Related issue #1199 was resolved in the [latest release v1.17.2](https://github.com/aws-amplify/amplify-android/releases/tag/release_v1.17.2). Would you be able to let us know if your issue persists with the update?
